opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -150,11 +151,11 @@ namespace Opm
             extractParallelGridInformationToISTL(simulator_.vanguard().grid(), parallelInformation_);
 
             // For some reason simulator_.model().elementMapper() is not initialized at this stage
-            // Hence const auto& elemMapper = simulator_.model().elementMapper(); does not work.
+            const auto& elemMapper = simulator_.model().elementMapper(); //does not work.
             // Set it up manually
-            using ElementMapper =
-                Dune::MultipleCodimMultipleGeomTypeMapper<GridView>;
-            ElementMapper elemMapper(simulator_.vanguard().grid().leafGridView(), Dune::mcmgElementLayout());
+            //using ElementMapper =
+            //    Dune::MultipleCodimMultipleGeomTypeMapper<GridView>;
+            //ElementMapper elemMapper(simulator_.vanguard().gridView(), Dune::mcmgElementLayout());
             detail::findOverlapAndInterior(simulator_.vanguard().grid(), elemMapper, overlapRows_, interiorRows_);
```

Comment:
Why is the leafGridView passed here and not the gridView?
`ElementMapper elemMapper(simulator_.vanguard().gridView(), Dune::mcmgElementLayout());`
is also working fine together with dune-fem

Reply:
Because the grid view stored in the vanguard is the LeafGridView of the grid. For the grid you always have to choose  between leaf or level grid view.

Reply:
Almost. There is a #if HAVE_DUNE_FEM in the basevanguard that makes a difference.

Reply:
This I did not understanf

Reply:
Code from basevanguard.hh 
```
    void updateGridView_()
    {
#if HAVE_DUNE_FEM
        // first delete old grid part
        // this is due to a bug in dune-fem (dangling reference)
        gridPart_.reset();
        gridPart_.reset(new GridPart(asImp_().grid()));
        gridView_.reset(new GridView(static_cast<GridView>(*gridPart_)));
        assert(gridView_->size(0) == asImp_().grid().leafGridView().size(0));
#else
        gridView_.reset(new GridView(asImp_().grid().leafGridView()));
#endif
    }
```